src/extensions/default/InAppNotifications/main.js

```diff
@@ -0,0 +1,372 @@
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/**
+ *  module for displaying in-app notifications
+ *
+ */
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit              = brackets.getModule("utils/AppInit"),
+        Dialogs              = brackets.getModule("widgets/Dialogs"),
+        DefaultDialogs       = brackets.getModule("widgets/DefaultDialogs"),
+        PreferencesManager   = brackets.getModule("preferences/PreferencesManager"),
+        ExtensionUtils       = brackets.getModule("utils/ExtensionUtils"),
+        ExtensionManager     = brackets.getModule("extensibility/ExtensionManager"),
+        Strings              = brackets.getModule("strings"),
+        Mustache             = brackets.getModule("thirdparty/mustache/mustache"),
+        HealthLogger         = brackets.getModule("utils/HealthLogger"),
+        NotificationBarHtml  = require("text!htmlContent/notificationContainer.html");
+
+    ExtensionUtils.loadStyleSheet(module, "styles/styles.css");
+
+    // duration of one day in milliseconds
+    var ONE_DAY = 1000 * 60 * 60 * 24;
+
+    // duration of two minutes in milliseconds
+    var TWO_MINUTES = 1000 * 60 * 2;
+
+    // Init default last notification number
+    PreferencesManager.stateManager.definePreference("lastHandledNotificationNumber", "number", 0);
+
+    // Init default last info URL fetch time
+    PreferencesManager.stateManager.definePreference("lastNotificationURLFetchTime", "number", 0);
+
+    /**
+     * Construct a new version update url with the given locale.
+     *
+     * @param {string=} locale - optional locale, defaults to 'brackets.getLocale()' when omitted.
+     * @param {boolean=} removeCountryPartOfLocale - optional, remove existing country information from locale 'en-gb' => 'en'
+     * return {string} the new version update url
+     */
+    function _getVersionInfoUrl(locale, removeCountryPartOfLocale) {
+
+        locale = locale || brackets.getLocale();
+
+        if (removeCountryPartOfLocale) {
+            locale = locale.substring(0, 2);
+        }
+
+        // PRERELEASE_BEGIN
+        // The following code is needed for supporting notifications in prerelease,
```

Comment:
Can you please help me understand how does this work only for prerelease?

Reply:
Removed the whole block.